src/widgets/TextManager.cpp

```diff
@@ -717,7 +717,7 @@ void TextManager::updateWindowCoord()
 
 void TextManager::insertTag(QAction *action)
 {
-	if (sender() != action->parentWidget()) {
+	if (sender() != qobject_cast<QObject*>(action->parent())) {
```

Comment:
Also an eye on this one, Qt suggested to go this way, but maybe it's not fully compatible with the check you were doing before?

Reply:
Here is the full QAction::parentWidget function:

```cpp
QWidget *QAction::parentWidget() const
{
    QObject *ret = parent();
    while (ret && !ret->isWidgetType())
        ret = ret->parent();
    return static_cast<QWidget*>(ret);
}
```

Reply:
Normally, sender() != action->parent() could be enough

Reply:
So do I need to cast it to QWidget* instead? However `sender()` returns a `QObject*` :( I'm confused

Reply:
you don't need to cast to QWidget, I think I use parentWidget instead of parent to ensure I target QToolBar or QMenu when I look for the parent of the QAction. I check if it is needed or not wait

Reply:
Okay I confirm, this is enough to work:

```suggestion
	if (sender() != action->parent()) {
```